src/utils/rdf.js

```diff
@@ -1,6 +1,75 @@
-import { Parser } from 'n3';
+// @ts-nocheck
+import { Parser, Store, DataFactory } from 'n3';
+import { newEngine } from '@comunica/actor-init-sparql-rdfjs';
 import { fromJsonldjsNode } from './claimgraph';
 
+/**
+ * Converts a a JSON-LD RDF object to N3 data type
+ * @param {object} node - JSON-LD RDF representation object
+ * @returns {any}
+ */
+export function toJsonldjsNode(node) {
+  if (node.DefaultGraph) {
+    return DataFactory.defaultGraph();
+  } else if (node.Iri) {
+    return DataFactory.namedNode(node.Iri);
+  } else if (node.Blank) {
+    return DataFactory.blankNode(node.Blank);
+  } else if (node.Literal) {
+    return DataFactory.literal(node.Literal.value, {
+      value: node.Literal.datatype,
+      language: node.Literal.language,
+    });
+  }
+
+  throw new Error(`Unable to determine node type: ${node}`);
+}
+
+/**
+ * Converts a claimgraph JSON representation into an N3 RDF store
+ * @param {object} claimgraph - JSON-LD RDF object
+ * @returns {Store}
+ */
+export function claimgraphToStore(claimgraph) {
+  const store = new Store();
+  claimgraph.forEach((quad) => {
+    const subject = quad[0];
+    const predicate = quad[1];
+    const object = quad[2];
+    const graph = quad.length > 3 ? toJsonldjsNode(quad[3]) : undefined;
+    store.addQuad(DataFactory.quad(
+      toJsonldjsNode(subject),
+      toJsonldjsNode(predicate),
+      toJsonldjsNode(object),
+      graph,
+    ));
+  });
+  return store;
+}
+
+/**
+ * Queries a claimgraph object and returns its bindings as an RDF term list
+ * @param {object} claimgraph - JSON-LD RDF object
+ * @param {string} query - SPARQL query string
+ * @param {object|null} engine - RDF query engine or null to auto-create
+ * @returns {array<any>}
+ */
+export async function queryClaimgraph(claimgraph, query, engine = null) {
+  // Create an N3 store from claimgraph JSON object
+  const store = typeof claimgraph === 'string' ? claimgraph : claimgraphToStore(claimgraph);
```

Comment:
nit: Arguments that can be one of a number of types makes me uncomfortable, especially when the behavior is not documented.

Reply:
nit: there doesn't seem to be any test that exercises the ability to pass claimgraph as a string.

Reply:
> nit: there doesn't seem to be any test that exercises the ability to pass claimgraph as a string.

wasn't aware that this was a required use case. what benefit is there to having this method support multiple types? i'd prefer that the string->obj conversion was done out of band. and by string, do you mean JSON string or some other format?

Reply:
It's not a required use case. I don't think there is a benefit to the method supporting multiple types. I also prefer that the string->obj conversion happen out of band.

This line is checking whether claimgraph is a string. Why is that?

Reply:
my bad, i misunderstood your first comments. so, this was initially intended to provide support for URI sources. you're right in that its not entirely clear from the method. this was done when first testing queries, i don't think it's actually needed to be included here - but that depends on if you think the crawler would need URI sources. my assumption is no.

Reply:
i have removed it, ill mark as resolved - lmk if you feel like its useful supporting more store types